lib/src/operation.rs

```diff
@@ -117,6 +118,19 @@ impl Operation {
         &self.data.metadata
     }
 
+    /// Returns true if predecessors are recorded in this operation.
+    ///
+    /// This returns false only if the operation was written by old jj.
+    pub fn stores_commit_predecessors(&self) -> bool {
+        self.data.commit_predecessors.is_some()
+    }
+
+    /// Returns predecessors of the specified commit if recorded.
+    pub fn predecessors_by_commit(&self, commit_id: &CommitId) -> Option<&[CommitId]> {
```

Comment:
nit: The name `predecessors_by_commit()` makes me think this returns a mapping of predecessors by commit. I would expect this to be called `predecessors_for_commit()`